LibreHardwareMonitorLib/Hardware/Gpu/AmdGpuGroup.cs

```diff
@@ -77,9 +90,43 @@ public AmdGpuGroup(ISettings settings)
                             _report.AppendLine(adapterInfo[i].FunctionNumber.ToString(CultureInfo.InvariantCulture));
                             _report.Append("AdapterID: 0x");
                             _report.AppendLine(adapterId.ToString("X", CultureInfo.InvariantCulture));
-
-                            if (!string.IsNullOrEmpty(adapterInfo[i].UDID) && adapterInfo[i].VendorID == AtiAdlxx.ATI_VENDOR_ID)
-                                potentialHardware.Add(new AmdGpu(adapterInfo[i], settings));
+                            _report.AppendLine("Family: " + gcnInfo.ASICFamilyId);
+
+                            int sensorsSupported = 0;
+                            if (AtiAdlxx.UsePmLogForFamily(gcnInfo.ASICFamilyId) &&
+                                AtiAdlxx.ADL_Method_Exists(nameof(AtiAdlxx.ADL2_Adapter_PMLog_Support_Get)) &&
+                                AtiAdlxx.ADL_Method_Exists(nameof(AtiAdlxx.ADL2_Device_PMLog_Device_Create)))
+                            {
+                                if (AtiAdlxx.ADLStatus.ADL_OK == AtiAdlxx.ADL2_Device_PMLog_Device_Create(_context, adapterInfo[i].AdapterIndex, ref device) &&
+                                    AtiAdlxx.ADLStatus.ADL_OK == AtiAdlxx.ADL2_Adapter_PMLog_Support_Get(_context, adapterInfo[i].AdapterIndex, ref pmLogSupportInfo))
+                                {
+                                    int k = 0;
+                                    while (pmLogSupportInfo.usSensors[k] != (ushort)AtiAdlxx.ADLPMLogSensors.ADL_SENSOR_MAXTYPES)
+                                    {
+                                        k++;
+                                    }
+                                    sensorsSupported = k;
+                                }
+                                _report.AppendLine("Sensors Supported: " + sensorsSupported);
+
+                                if (device != 0)
+                                {
+                                    AtiAdlxx.ADL2_Device_PMLog_Device_Destroy(_context, device);
+                                }
+                            }
+
+                            if (!string.IsNullOrEmpty(adapterInfo[i].UDID) && adapterInfo[i].VendorID == AtiAdlxx.ATI_VENDOR_ID &&
+                                !IsAdapterAlreadyFound(adapterInfo[i].BusNumber, adapterInfo[i].DeviceNumber))
```

Comment:
Is this ever returning the same adapter here?

Reply:
Yes, in my case I have 5, all of them have all sensors:

```
Number of adapters: 9

AdapterIndex: 0
isActive: 1
AdapterName: AMD Radeon RX 7800 XT
UDID: PCI_VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8_6&8C9B63C&0&00000008A
PNPString: PCI\VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8\6&8C9B63C&0&00000008
Present: 1
VendorID: 0x1002
BusNumber: 3
DeviceNumber: 0
FunctionNumber: 0
AdapterID: 0x35CAF060

AdapterIndex: 1
isActive: 0
AdapterName: AMD Radeon RX 7800 XT
UDID: PCI_VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8_6&8C9B63C&0&00000008&02A
PNPString: PCI\VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8\6&8C9B63C&0&00000008&02
Present: 1
VendorID: 0x1002
BusNumber: 3
DeviceNumber: 0
FunctionNumber: 0
AdapterID: 0x35CAF060

AdapterIndex: 2
isActive: 0
AdapterName: AMD Radeon RX 7800 XT
UDID: PCI_VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8_6&8C9B63C&0&00000008&03A
PNPString: PCI\VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8\6&8C9B63C&0&00000008&03
Present: 1
VendorID: 0x1002
BusNumber: 3
DeviceNumber: 0
FunctionNumber: 0
AdapterID: 0x35CAF060

AdapterIndex: 3
isActive: 0
AdapterName: AMD Radeon RX 7800 XT
UDID: PCI_VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8_6&8C9B63C&0&00000008&04A
PNPString: PCI\VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8\6&8C9B63C&0&00000008&04
Present: 1
VendorID: 0x1002
BusNumber: 3
DeviceNumber: 0
FunctionNumber: 0
AdapterID: 0x35CAF060

AdapterIndex: 4
isActive: 0
AdapterName: AMD Radeon RX 7800 XT
UDID: PCI_VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8_6&8C9B63C&0&00000008&05A
PNPString: PCI\VEN_1002&DEV_747E&SUBSYS_05FD1043&REV_C8\6&8C9B63C&0&00000008&05
Present: 1
VendorID: 0x1002
BusNumber: 3
DeviceNumber: 0
FunctionNumber: 0
AdapterID: 0x35CAF060
```